request.cpp

```diff
@@ -0,0 +1,23 @@
+#include "request.h"
+
+Request::Request(const std::string& start_str)
+    : m_verb(start_str.substr(0, 3)),
```

Comment:
If the verb is `POST` or `DELETE` your `m_verb` will contain `POS` or `DEL`, which is not correct.